openfl-tutorials/interactive_api/HPU/PyTorch_MedMNIST_2D/workspace/HPU_Pytorch_MedMNIST_2D.ipynb

```diff
@@ -0,0 +1,622 @@
+   "id": "4af5c4c2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fed_dataset = MedMnistFedDataset(train_bs=TRAIN_BS, valid_bs=VALID_BS)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7f63908e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fed_dataset.shard_descriptor = dummy_shard_desc\n",
+    "for i, (sample, target) in enumerate(fed_dataset.get_train_loader()):\n",
+    "    print(sample.shape, target.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "075d1d6c",
+   "metadata": {},
+   "source": [
+    "### Describe the model and optimizer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8477a001",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define a simple CNN model\n",
+    "class Net(nn.Module):\n",
+    "    def __init__(self, in_channels, num_classes):\n",
+    "        super(Net, self).__init__()\n",
+    "\n",
+    "        self.layer1 = nn.Sequential(\n",
+    "            nn.Conv2d(in_channels, 16, kernel_size=3),\n",
+    "            nn.BatchNorm2d(16),\n",
+    "            nn.ReLU())\n",
+    "\n",
+    "        self.layer2 = nn.Sequential(\n",
+    "            nn.Conv2d(16, 16, kernel_size=3),\n",
+    "            nn.BatchNorm2d(16),\n",
+    "            nn.ReLU(),\n",
+    "            nn.MaxPool2d(kernel_size=2, stride=2))\n",
+    "\n",
+    "        self.layer3 = nn.Sequential(\n",
+    "            nn.Conv2d(16, 64, kernel_size=3),\n",
+    "            nn.BatchNorm2d(64),\n",
+    "            nn.ReLU())\n",
+    "        \n",
+    "        self.layer4 = nn.Sequential(\n",
+    "            nn.Conv2d(64, 64, kernel_size=3),\n",
+    "            nn.BatchNorm2d(64),\n",
+    "            nn.ReLU())\n",
+    "\n",
+    "        self.layer5 = nn.Sequential(\n",
+    "            nn.Conv2d(64, 64, kernel_size=3, padding=1),\n",
+    "            nn.BatchNorm2d(64),\n",
+    "            nn.ReLU(),\n",
+    "            nn.MaxPool2d(kernel_size=2, stride=2))\n",
+    "\n",
+    "        self.fc = nn.Sequential(\n",
+    "            nn.Linear(64 * 4 * 4, 128),\n",
+    "            nn.ReLU(),\n",
+    "            nn.Linear(128, 128),\n",
+    "            nn.ReLU(),\n",
+    "            nn.Linear(128, num_classes))\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = self.layer1(x)\n",
+    "        x = self.layer2(x)\n",
+    "        x = self.layer3(x)\n",
+    "        x = self.layer4(x)\n",
+    "        x = self.layer5(x)\n",
+    "        x = x.view(x.size(0), -1)\n",
+    "        x = self.fc(x)\n",
+    "        return x\n",
+    "\n",
+    "# NOTE: This is a workaround to move model to HPU before initializing any optimizers \n",
+    "device = torch.device('hpu')\n",
+    "model = Net(in_channels=n_channels, num_classes=n_classes).to(device)\n",
+    "    \n",
+    "# define loss function and optimizer\n",
+    "if task == \"multi-label, binary-class\":\n",
```

Comment:
I do not get what you intended to do with this code. You must understand that the notebook is not executed remotely, please place all the logic inside OpenFL interface classes

Reply:
This was part of original interactive api [medmnist code](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_MedMNIST_2D/workspace). But currently for HPU example I have modified it according to your suggestion